instrumentor/instrumentor.go

```diff
@@ -37,6 +49,41 @@ func New(opts controllers.KubeManagerOptions, dp *distros.Provider) (*Instrument
 	// This code can be removed once the migration is confirmed to be successful.
 	mgr.Add(&runtimemigration.MigrationRunnable{KubeClient: mgr.GetClient(), Logger: opts.Logger})
 
+	// setup the certificate rotator
+	rotatorSetupFinished := make(chan struct{})
+	err = rotator.AddRotator(mgr, &rotator.CertRotator{
+		SecretKey: types.NamespacedName{
+			Namespace: env.GetCurrentNamespace(),
+			Name:      k8sconsts.InstrumentorWebhookSecretName,
+		},
+		CertDir: filepath.Join(os.TempDir(), "k8s-webhook-server", "serving-certs"),
+		IsReady: rotatorSetupFinished,
+		CAName:  k8sconsts.InstrumentorCAName,
+		Webhooks: []rotator.WebhookInfo{
+			{Name: k8sconsts.InstrumentorMutatingWebhookName, Type: rotator.Mutating},
+			{Name: k8sconsts.InstrumentorSourceMutatingWebhookName, Type: rotator.Mutating},
+			{Name: k8sconsts.InstrumentorSourceValidatingWebhookName, Type: rotator.Validating},
+		},
+		DNSName: "serving-cert",
+		ExtraDNSNames: []string{
+			fmt.Sprintf("%s.%s.svc", k8sconsts.InstrumentorServiceName, env.GetCurrentNamespace()),
+			fmt.Sprintf("%s.%s.svc.cluster.local", k8sconsts.InstrumentorServiceName, env.GetCurrentNamespace()),
+		},
+		EnableReadinessCheck: true,
+
+		// we could set RequireLeaderElection to true here but that will make the readiness probe fail for non-leader
+		// instances (since the IsReady channel will not be closed in non-leader instances).
+
+		// these are the defaults, but we set them explicitly for clarity
+		CaCertDuration:         10 * 365 * 24 * time.Hour, // 10 years
+		ServerCertDuration:     1 * 365 * 24 * time.Hour,  // 1 year
+		RotationCheckFrequency: 12 * time.Hour,            // 12 hours
+		LookaheadInterval:      90 * 24 * time.Hour,       // 90 days
+	})
```

Comment:
can we share this part which is duplicated in instroumentor and autoscaler via k8sutils?